lib/process/process.go

```diff
@@ -621,6 +621,67 @@ func (p *Process) runRegistrySynchronizer(ctx context.Context) {
 	}
 }
 
+func (p *Process) reconcileNodeLabels(client *kubernetes.Clientset) error {
+	cluster, err := p.operator.GetLocalSite()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	nodes, err := utils.GetNodes(client.CoreV1().Nodes())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for ip, node := range nodes {
+		server, err := cluster.ClusterState.FindServerByIP(ip)
+		if err != nil {
+			return trace.Wrap(err)
```

Comment:
I wonder what this error actually means? It can mean borked cluster state or unexpectedly added kubernetes node?

Reply:
Yeah, not sure how that can actually happen but the practice shows that everything that can go wrong, will. I've modified this loop a bit so the failure to reconcile a node is just logged and does not stop the entire loop.